agent/dns.go

```diff
@@ -501,7 +512,14 @@ func (d *DNSServer) addSOA(cfg *dnsConfig, msg *dns.Msg) {
 // in the current cluster which serve as authoritative name servers for zone.
 
 func (d *DNSServer) nameservers(cfg *dnsConfig, maxRecursionLevel int) (ns []dns.RR, extra []dns.RR) {
-	out, err := d.lookupServiceNodes(cfg, d.agent.config.Datacenter, structs.ConsulServiceName, "", structs.DefaultEnterpriseMeta(), false)
+	lookup := serviceLookup{
+		Datacenter:     d.agent.config.Datacenter,
+		Service:        structs.ConsulServiceName,
+		Connect:        false,
+		Ingress:        false,
+		EnterpriseMeta: *structs.DefaultEnterpriseMeta(),
+	}
+	out, err := d.lookupServiceNodes(cfg, lookup)
```

Comment:
You don't have to define the `lookup` variable at all, you could just inline it:
```
out, err := d.lookupServiceNodes(cfg, serviceLookup{
   ...stuff...
})
```

I did something like this in the discovery chain compiler and it's not too bad. It ends up feeling a bit like python kwargs.